graphics/gpick/Portfile

```diff
@@ -18,41 +20,27 @@ long_description        Gpick is a program used to pick colors from anywhere \
                         shades and tints and export palettes to common file \
                         formats or simply copy them to the clipboard.
 
-homepage                https://code.google.com/p/gpick/
-master_sites            googlecode
-distname                ${name}_${version}
+github.tarball_from     releases
 
-checksums               rmd160  a7c65539780b1eb74f44538b165dbff923762beb \
-                        sha256  d57dcbc55dcd1b10a103de8ae6a7028063a4ec8934372c587107dd772da5397f \
-                        size    337112
+checksums               rmd160  19ebed9cb8229e319d8b68013b46c87a6842cffc \
+                        sha256  90b03ce10d2f3e211a3f37d262596e4ae40d05cbd633aa0560d0891562c93474 \
+                        size    368100
 
-boost.version           1.71
```

Comment:
Shouldn't every port that uses the boost portgroup specify which `boost.version` to use, since the point is that boost versions are often not compatible with one another?

I was going with 1.80 in the update I was working on, since trying to use 1.81 produced a warning message.

Reply:
I am not well versed in Boost version incompatibilities.
I just assumed the default boost version was somehow significant, especially since the default Boost version is different for different OS versions.
I am more that happy to set a different version of Boost if that is more appropriate.

Reply:
> I am not well versed in Boost version incompatibilities.

Back when we only had the one boost port, practically every time we updated it to a new major version, it would break some port. The developers of boost do not maintain backward compatibility in new versions of boost; they [claim](https://www.boost.org/users/faq.html) this "isn't always possible" and recommend "freezing the version of the Boost libraries used by your project". Projects should probably bundle whatever version of boost they are compatible with. For those that don't, we now have many boost ports so each port can depend on whichever version is best for that project.

> I just assumed the default boost version was somehow significant, especially since the default Boost version is different for different OS versions.

Well that's true, I'm not sure what to do about the different default boost versions on different OS versions though it sounds like we want to get rid of that difference; see the [comments on this commit](https://github.com/macports/macports-ports/commit/e5316596d177c9f6370d5817f94b94879a939405). I think which boost version the boost portgroup uses as its default is not necessarily significant anymore. It's not something we can ever change anymore—not without also either revbumping and testing all ports that don't specify the boost version, or changing them all to specify the formerly-default boost version. In my opinion it's an argument for having the boost portgroup not specify any default boost version and requiring each port to specify it. See [ticket 66927](https://trac.macports.org/ticket/66927).

Reply:
> I was going with 1.80 in the update I was working on, since trying to use 1.81 produced a warning message.

My motivation in using a newer boost was that I remembered there were tickets like [66927](https://trac.macports.org/ticket/66927) where it had been requested that we update ports to newer boost versions since old boost versions did not build on some systems. However I hadn't remembered that 1.81 was the version in which that problem was fixed.

When trying to use 1.81, cmake printed this warning:

```
CMake Warning at /opt/local/share/cmake-3.24/Modules/FindBoost.cmake:1384 (message):
  New Boost version may have incorrect or missing dependencies and imported
  targets
```

When I looked this up, I understood that cmake prints this warning when the boost version is newer than was available at the time that this version of cmake was released. I guess boost 1.80 was the latest version available when cmake 3.24.4 was released, and I guess there's the possibility that new boost versions need changes to the cmake files to work properly. But if we want to try using 1.81 and it works fine, we can add `-DBoost_NO_WARN_NEW_VERSIONS=ON` to configure.args to suppress the warning or maybe we should [update cmake](https://trac.macports.org/ticket/67540) first.

Looks like a [small change for boost 1.81 support](https://github.com/Kitware/CMake/commit/a97032c4e7b1efb4c5535e5f705ef8ffd5fc2dd3) was made in cmake 3.25.2 and there was another [small change for boost 1.82 support](https://github.com/Kitware/CMake/commit/5cbbe55de85bdd0d6d90241a9b18683f13375fc1) committed to cmake a couple hours ago and not yet part of any release.

Reply:
I update the PR to include `boost.version 1.80` since, as you say, it is the latest version to produce no warnings.
Since Boost is only a build dependency, it should be relatively easy to increase the version if it is decided that a more recent Boost version works as well.